### ISSUE TITLE
Getting deprecated warning, while creating standard cluster

### ISSUE THREAD

**Issue by Dipanshi293** [GoogleCloudPlatform/cloud-foundation-fabric#2349]
**Describe the bug**
Hi Team,

We are calling CFF module and resource "google_container_cluster", when we are creating standard cluster, we are getting this warning:
```
**Warning: Argument is deprecated**
with module.kubernetes-engine_gke-standard-cluster.module.gke_cluster.google_container_cluster.cluster
on .terraform/modules/kubernetes-engine_gke-standard-cluster.gke_cluster/modules/gke-cluster-standard/main.tf line 374, in resource "google_container_cluster" "cluster":
  monitoring_config {
Deprecated in favor of enable_relay field. Remove this attribute's configuration as this field will be removed in the next major release and enable_relay will become a required field.
```
And we have not declared this attribute(enable_relay) in our module.

**Main.tf**
```
module "gke_cluster" {
  source              = "git::https://github.com/GoogleCloudPlatform/cloud-foundation-fabric.git//modules/gke-cluster-standard?ref=v31.0.0"
  name                = var.cluster_name
  project_id          = var.project_id
  location            = var.region
  labels              = var.labels
  deletion_protection = var.deletion_protection
  vpc_config = {
    network    = data.google_compute_subnetwork.default.network
    subnetwork = data.google_compute_subnetwork.default.self_link
    secondary_range_names = {
      pods     = var.secondary_range_pods
      services = var.secondary_range_services
    }
    master_ipv4_cidr_block   = var.master_ipv4_cidr_block
    master_authorized_ranges = var.master_authorized_ranges
  }

  private_cluster_config = {
    enable_private_nodes    = true
    enable_private_endpoint = true
    master_global_access    = false
  }

  enable_addons = {
    cloudrun                       = var.cloudrun_addon
    config_connector               = false
    dns_cache                      = true
    horizontal_pod_autoscaling     = true
    http_load_balancing            = true
    gce_persistent_disk_csi_driver = true
    gcp_filestore_csi_driver       = true
  }

  enable_features = {
    binary_authorization = true
    cost_management      = true
    shielded_nodes       = true
    dns = var.dns_provider != "PLATFORM_DEFAULT" ? {
      provider = var.dns_provider
      scope    = var.dns_scope
      domain   = var.dns_domain
    } : null
    database_encryption = {
      state    = "ENCRYPTED"
      key_name = var.database_encryption_key_name
    }
    workload_identity    = true
    l4_ilb_subsetting    = true
    groups_for_rbac      = var.groups_for_rbac
    dataplane_v2         = true
    gateway_api          = true
    intranode_visibility = true
    network_policy       = true
  }

  cluster_autoscaling = var.cluster_autoscaling

  node_config = {
    tags              = var.tags
    service_account   = var.service_account
    boot_disk_kms_key = var.boot_disk_kms_key
  }
  issue_client_certificate = false
  release_channel          = var.release_channel
  max_pods_per_node        = var.max_pods_per_node

  # Logging and monitoring
  logging_config = {
    enable_api_server_logs         = true
    enable_scheduler_logs          = true
    enable_controller_manager_logs = true
    enable_system_logs             = true
    enable_workloads_logs          = true
  }

  monitoring_config = {
    enable_system_metrics = true
    # Control plane metrics
    enable_api_server_metrics         = true
    enable_controller_manager_metrics = true
    enable_scheduler_metrics          = true
    # Kube state metrics
    enable_daemonset_metrics   = true
    enable_deployment_metrics  = true
    enable_hpa_metrics         = true
    enable_pod_metrics         = true
    enable_statefulset_metrics = true
    enable_storage_metrics     = true
    # Advanced datapath observability metrics 
    advanced_datapath_observability = {
      enable_metrics = var.advanced_datapath_observability_enable_metrics
      relay_mode     = var.advanced_datapath_observability_relay_mode
    }
    # Google Cloud Managed Service for Prometheus
    enable_managed_prometheus = var.enable_managed_prometheus
  }

  # Backup for GKE
  backup_configs = var.backup_configs
}
```

**Comment by ludoo** [comment]
Thanks for flagging this, I created a couple clusters 3-4 days ago and did not have this warning. Must be very recent.

We usually track the provider, it probably makes sense to wait until the provider deprecates that field before making changes.

**Comment by ludoo** [comment]
This was fixed in #2366 (provider PR that deprecated the field is [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/9633)).
